### PR TITLE
feat(deps): Update Elasticsearch plugin-sdk to v1.38.2

### PR DESCRIPTION
**Pull request by erezrokah**
#### Summary

Extracted from https://github.com/cloudquery/cloudquery/pull/8156 as it requires an additional fix.
If the table doesn't have any PKs and in overwrite mode(s) we can only write a single resource per table as the doc ID will always be the same.
This PR changes it so if there are no PKs, we use all columns for the doc ID.

> Discovered by the `MigrateOverwrite` test

### CODE REVIEW

File: plugins/destination/elasticsearch/client/client_test.go [hunk 1]
```diff
@@ -22,5 +31,8 @@ func TestPlugin(t *testing.T) {
 		destination.PluginTestSuiteTests{
 			SkipMigrateOverwriteForce: true,
 			SkipMigrateAppendForce:    true,
+
```
Comment (erezrokah) [inline plugins/destination/elasticsearch/client/client_test.go:34]: Once we add force migrate mode we can remove the `SkipMigrateOverwriteForce` and `SkipMigrateAppendForce`, but this is out of scope of this PR

File: plugins/destination/elasticsearch/client/write.go [hunk 2]
```diff
@@ -62,6 +62,12 @@ func (c *Client) WriteTableBatch(ctx context.Context, table *schema.Table, resou
 
 func pkIndexes(table *schema.Table) []int {
 	pks := table.PrimaryKeys()
+	if len(pks) == 0 {
+		// if no PK is defined, use all columns for the ID which is based on the indices returned by this function
+		for _, col := range table.Columns {
+			pks = append(pks, col.Name)
+		}
```
Comment (candiduslynx) [inline plugins/destination/elasticsearch/client/write.go:69]:
```suggestion
		for _, col := range table.Columns {
			pks = table.Columns.Names()
		}
```

**Review by candiduslynx — CHANGES_REQUESTED** [review]

File: plugins/destination/elasticsearch/client/write.go (hunk 2 above)
Comment (erezrokah) [reply plugins/destination/elasticsearch/client/write.go:69]: Nice one 🚀

Comment (candiduslynx) [reply plugins/destination/elasticsearch/client/write.go]: 100%

**Review by candiduslynx — APPROVED** [review]
I'm approving but removing automerge bc the original author (@hermanschaaf) should take a look, too.

**Review by hermanschaaf — APPROVED** [review]
Nice catch!

**Pull request merged**